datasources/src/main/java/org/opensearch/sql/datasources/storage/OpenSearchDataSourceMetadataStorage.java

```diff
@@ -217,6 +217,8 @@ private List<DataSourceMetadata> searchInDataSourcesIndex(QueryBuilder query) {
     searchSourceBuilder.query(query);
     searchSourceBuilder.size(DATASOURCE_QUERY_RESULT_SIZE);
     searchRequest.source(searchSourceBuilder);
+    // strongly consistent reads is requred. more info https://github.com/opensearch-project/sql/issues/1801.
+    searchRequest.preference("_primary");
```

Comment:
@penghuo : Is there any data around TPS on this search query ?  I am thinking if this can have any performance implications.